internal/qldb/fill_order.go

```diff
@@ -0,0 +1,337 @@
+/**
+ * Copyright 2023-present Coinbase Global, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package qldb
+
+import (
+	"context"
+	"fmt"
+	"math/big"
+	"strings"
+	"time"
+
+	"github.com/coinbase-samples/ib-ledger-go/internal/utils"
+	api "github.com/coinbase-samples/ib-ledger-go/pkg/pbs/ledger/v1"
+	"github.com/google/uuid"
+
+	"github.com/amzn/ion-go/ion"
+	"github.com/awslabs/amazon-qldb-driver-go/v3/qldbdriver"
+	"github.com/coinbase-samples/ib-ledger-go/internal/config"
+	"github.com/coinbase-samples/ib-ledger-go/internal/model"
+)
+
+func FillOrder(req *api.PostFillRequest) error {
+	_, err := Repo.Driver.Execute(
+		context.Background(),
+		func(txn qldbdriver.Transaction) (interface{}, error) {
```

Comment:
This function is really long and hard to read. Can you break it up into more discrete components?